docs/essentials/gc/unmanaged/implementingdispose.md

```diff
@@ -1 +1,442 @@
+
+Here's the general pattern for implementing the dispose pattern for a base class that uses a safe handle. 
+
+```csharp
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+class BaseClass : IDisposable
+{
+   // Flag: Has Dispose already been called?
+   bool disposed = false;
+   // Instantiate a SafeHandle instance.
+   SafeHandle handle = new SafeFileHandle(IntPtr.Zero, true);
+
+   // Public implementation of Dispose pattern callable by consumers.
+   public void Dispose()
+   { 
+      Dispose(true);
+      GC.SuppressFinalize(this);           
+   }
+
+   // Protected implementation of Dispose pattern.
+   protected virtual void Dispose(bool disposing)
+   {
+      if (disposed)
+         return; 
+
+      if (disposing) {
+         handle.Dispose();
+         // Free any other managed objects here.
+         //
+      }
+
+      // Free any unmanaged objects here.
+      //
+      disposed = true;
+   }
+}
+```
+
+> **Note**
+>
+> The previous example uses a [SafeFileHandle]()http://dotnet.github.io/api/Microsoft.Win32.SafeHandles.SafeFileHandle.html object to illustrate the pattern; any object derived from `SafeHandle` could be used instead. Note that the example does not properly instantiate its `SafeFileHandle` object. 
+
+Here's the general pattern for implementing the dispose pattern for a base class that overrides `Object.Finalize`. 
+
+```csharp
+using System;
+
+class BaseClass : IDisposable
+{
+   // Flag: Has Dispose already been called?
+   bool disposed = false;
+
+   // Public implementation of Dispose pattern callable by consumers.
+   public void Dispose()
+   { 
+      Dispose(true);
+      GC.SuppressFinalize(this);           
+   }
+
+   // Protected implementation of Dispose pattern.
+   protected virtual void Dispose(bool disposing)
+   {
+      if (disposed)
+         return; 
+
+      if (disposing) {
+         // Free any other managed objects here.
+         //
+      }
+
+      // Free any unmanaged objects here.
+      //
+      disposed = true;
+   }
+
+   ~BaseClass()
+   {
+      Dispose(false);
+   }
+}
+```
+
+> **Note**
+>
+> In C#, you override `Object.Finalize` by defining a `destructor`.
+
+## Implementing the dispose pattern for a derived class
+
+A class derived from a class that implements the `IDisposable` interface shouldn't implement `IDisposable`, because the base class implementation of `IDisposable.Dispose` is inherited by its derived classes. Instead, to implement the dispose pattern for a derived class, you provide the following: 
+
+* A `protected Dispose(Boolean)` method that overrides the base class method and performs the actual work of releasing the resources of the derived class. This method should also call the `Dispose(Boolean)` method of the base class and pass it a value of `true` for the *disposing* argument. 
```

Comment:
> This method should also call the `Dispose(Boolean)` method of the base class and pass it a value of `true` for the *disposing* argument.

It shouldn't pass `true` and the two examples below don't, it should pass the value of `disposing` it received.